### ISSUE TITLE
Not working with vitejs in production mode

### ISSUE THREAD

**Issue by jrmi** [Mastermindzh/react-cookie-consent#110]
I tried to use `react-cookie-consent` in a `vitejs` app. It works in development mode but in production, react throws me this error:

```
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.
```

I've joined a test project to reproduce it. [test-cookie.tar.gz](https://github.com/Mastermindzh/react-cookie-consent/files/6152221/test-cookie.tar.gz). 

It's a vitejs project from scratch with the minimal modifications (importing and using the lib) to reproduce the bug.

You can test it by executing:

```
npm ci
npm run dev # It's working on localhost:3000
npm run build 
npm run serve # It's broken on localhost:5000
```
I suspect a compatibility issue with imports...

Thanks for your project.

**Comment by Mastermindzh** [comment]
Hey @jrmi ,

I have no experience with Vitejs but I will try to find out what is wrong next week.
Don't have much time left this week, sorry :)

**Comment by lukebailey** [comment]
was there any progress on this, wea re seeing the same issue with ViteJS in production (works fine in DEV mode)

**Comment by Mastermindzh** [comment]
Not really.
I tried it and I for some reason thought it was because it was an older-style react Component instead of a functional component so I did a rewrite to a functional component but that threw the error:

```sh
Uncaught Error: Objects are not valid as a React child (found: object with keys {}). If you meant to render a collection of children, use an array instead
```

So for now I'm calling it quits. Might very well finish the rewrite sometime in the near future but it currently has no priority for me, sorry.

**Comment by jclohmann** [comment]
Hi! I am also kind of stuck in this problem.
There seems to be a problem with the generated default-export. Logging the imported component gave me this output:
```
import CookieConsent from 'react-cookie-consent';

console.log(CookieConsent);
```
Which gave me this output just before the described error:
```
Module {…}
	Cookies: ƒ o()
	OPTIONS: Object
	SAME_SITE_OPTIONS: Object
	default: ƒ i(e)
	getCookieConsentValue: ƒ ()
	Symbol(Symbol.toStringTag): "Module"
	__esModule: true
	get Cookies: ƒ ()
	get OPTIONS: ƒ ()
	get SAME_SITE_OPTIONS: ƒ ()
	get default: ƒ ()
	get getCookieConsentValue: ƒ ()
```
It seems the CookieConsent-component is not available as default but as a property named `default`.

As a workaround one can use the `default` properties value as a component by doing this:
```
import CookieConsentX from 'react-cookie-consent';

// @ts-ignore
const CookieConsent = CookieConsentX.default;

const SomeReactComponent = () => {
	return (
		<CookieConsent ... />
	);
};
```
This is not ideal when using typescript, since this replaces the correct type with `any`. (Which can surely be fixed using `as`.)

Another way I found was to change react-cookie-consent's webpack-config like this:
```
-    // libraryTarget: "commonjs2", // THIS IS THE MOST IMPORTANT LINE! :mindblow: I wasted more than 2 days until realize this was the line most important in all this guide.
+    library: {
+      name: "reactCookieConsent",
+      type: 'umd'
+    },
```
It works as expected with this config.

@Mastermindzh Do you think we can change the library-type to umd without issues?

**Comment by Mastermindzh** [comment]
Yes, and @peterkrieg made the PR which was just merged in [7.0.0](https://www.npmjs.com/package/react-cookie-consent/v/7.0.0)

Have fun!

**Comment by jrmi** [comment]
> Yes, and @peterkrieg made the PR which was just merged in [7.0.0](https://www.npmjs.com/package/react-cookie-consent/v/7.0.0)
> 
> Have fun!

Cool, thanks @peterkrieg !
